src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,698 @@
+
+    /// @dev Token decimals of the Orchestrator token
+    /// @notice Number of decimal places used by the collateral token for proper decimal handling
+    uint8 private _collateralTokenDecimals;
+
+    /// @dev Maximum fee that can be charged for sell operations, in basis points
+    /// @notice Maximum allowed project fee percentage for selling tokens
+    uint private _maxSellFee;
+
+    /// @dev Maximum fee that can be charged for buy operations, in basis points
+    /// @notice Maximum allowed project fee percentage for buying tokens
+    uint private _maxBuyFee;
+
+    /// @dev Order ID counter for tracking individual orders
+    /// @notice Unique identifier for the current order being processed
+    uint private _orderId;
+
+    /// @dev Counter for generating unique order IDs
+    /// @notice Keeps track of the next available order ID to ensure uniqueness
+    uint private _nextOrderId;
+
+    /// @dev Total amount of collateral tokens currently in redemption process
+    /// @notice Tracks the sum of all pending redemption orders
+    uint private _openRedemptionAmount;
+
+    /// @dev Flag indicating if direct operations are only allowed
+    bool private _isDirectOperationsOnly;
+
+    /// @dev Storage gap for future upgrades
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    //--------------------------------------------------------------------------
+    // Initialization Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        // Initialize parent contracts
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode config data
+        (
+            address oracleAddress_,
+            address issuanceToken_,
+            address acceptedToken_,
+            uint buyFee_,
+            uint sellFee_,
+            uint maxSellFee_,
+            uint maxBuyFee_,
+            bool isDirectOperationsOnly_
+        ) = abi.decode(
+            configData_,
+            (address, address, address, uint, uint, uint, uint, bool)
+        );
+
+        // Set accepted token
+        _token = IERC20(acceptedToken_);
+
+        // Cache token decimals for collateral
+        _collateralTokenDecimals = IERC20Metadata(address(_token)).decimals();
+
+        if (
+            !ERC165Upgradeable(address(oracleAddress_)).supportsInterface(
+                type(IOraclePrice_v1).interfaceId
+            )
+        ) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_InvalidOracleInterface(
+            );
+        }
+        // Set oracle
+        _oracle = IOraclePrice_v1(oracleAddress_);
+
+        // Initialize base functionality (should handle token settings)
+        _setIssuanceToken(issuanceToken_);
+
+        // Set fees (checking max fees)
```

Comment:
The actual settings happens a few lines later, so this is just checking (regarding the comment).

Reply:
rephrased